### PR TITLE
added bed comp option for G29

### PR DESCRIPTION
**Pull request by carlsoar**
G29.1 no longer allows bed compensation by default.
G29.2 allows bed compensation, but it can be turned off.

### CODE REVIEW

**Comment by carlsoar** [comment]
Elias can you look through this and make sure it's doing what it should? Haven't had time to check it, will run tonight if possible.

**Comment by eliasbakken** [comment]
Hey I'm at OHS in Denver this coming week, so check with Jon or Wiley to
see if they have some bandwidth to review the pull :)

**Comment by ajmirsky** [comment]
does G29.1 no longer allow bed compensation? or, it is allowed but now just disabled by default?

File: redeem/gcodes/G29.py [hunk 1]
```diff
@@ -104,11 +106,12 @@ def execute(self, g):
...
-        gcodes += "    M561 S; Show the current matrix\n"
+        if bed_comp:
+            gcodes += "    M561 U; (RFS) Update the matrix based on probe data\n"
+            gcodes += "    M561 S; Show the current matrix\n"
         gcodes += "    M500; (RFS) Save data\n"
 
-        # REset probe points
+        # Reset probe points
         self.printer.probe_points = []
         self.printer.config.set("Macros", "G29", gcodes)
```
Comment (ajmirsky) [inline redeem/gcodes/G29.py]: i think this means that if i had written a custom G29 and then executed G29.1, I'd lose my macro? maybe we should make a note in the description that executing 29.1/2 will over write the G29 macro in the local.cfg (?)?

**Comment by carlsoar** [comment]
29.1 defaults to off, oops, yeah mistake in the section there. 

Yes, G29 has always been overwritten by the use of G29.1 and 29.2, that was me fixing that capitalization in the original code.

I will make that change in the description.

Thanks!

**Pull request merged**